MaxText/layers/attentions.py

```diff
@@ -375,9 +384,14 @@ def move_kvlen_axis(self, kv):
     Returns:
       reshaped kv as [b, ..., n, d, s]
     """
-    return jax.numpy.moveaxis(kv, (0, 1, 2, 3), (2, 0, 1, 3))
+    axis_order_to_index_mapping = {a:i for i, a in enumerate(cached_axis_order)}
+    axis_destination = tuple([i for a, i in sorted(axis_order_to_index_mapping.items())])
```

Comment:
This different from the previous logic, which directly uses `cached_axis_order` and not the indices of sorted axes. For default value of (2, 0, 1, 3), code will use (1, 2, 0, 3) instead. Didn't this error out during your testing?

Reply:
To change the source axis 0,1,2,3 (logical layout's original axis) to 1,2,0,3 (default layout using original axis but reordered):
- axis 0 will move the 3rd position, i.e. destination axis 2
- axis 1 will move the 1st position, i.e. destination axis 0;  
- axis 2 will move the 2nd position, i.e. destination axis 1;  
- axis 3 will stay, i.e. destination axis 3;

Thus destination axis is 2,0,1,3